CharlieBackend.AdminPanel/HtmlHelpers/CalendarHelper.cs

```diff
@@ -0,0 +1,209 @@
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Encodings.Web;
+
+namespace CharlieBackend.AdminPanel.HtmlHelpers
+{
+    public static class CalendarHelper
+    {
+        public const int DaysInOneWeek = 7;
+        public static HtmlString CalendarBodyHtml(this IHtmlHelper html, CalendarViewModel calendar)
+        {
+            DateTime start, finish;
+
+            var eventOccurencesFiltered = calendar.ScheduledEvents.Select(x => calendar.EventOccurences.First(y => y.Id == x.EventOccuranceId)).ToList();
+
+            if (calendar.ScheduledEventFilter.FinishDate.HasValue)
+            {
+                finish = calendar.ScheduledEventFilter.FinishDate.Value;
+            }
+            else
+            {
+                DateTime latestFinish = eventOccurencesFiltered.Max(x => x.EventFinish);
+                DateTime latestStart = eventOccurencesFiltered.Max(x => x.EventStart);
+
+                finish = latestFinish > latestStart ? latestFinish : latestStart;
+            }
+
+            if (calendar.ScheduledEventFilter.StartDate.HasValue)
+            {
+                start = calendar.ScheduledEventFilter.StartDate.Value;
+            }
+            else
+            {
+                DateTime earliestFinish = eventOccurencesFiltered.Min(x => x.EventFinish);
+                DateTime earliestStart = eventOccurencesFiltered.Min(x => x.EventStart);
+
+                start = earliestFinish > earliestStart ? earliestStart : earliestFinish;
+            }
+
+            int daysCount = (int)(finish.Date - start.Date).TotalDays;
+
+            List<TagBuilder> dayContainers = new List<TagBuilder>();
+
+            for (int day = 1; day <= daysCount; day++)
+            {
+                dayContainers.Add(GetDayContainerHtml(start.AddDays(day), calendar.ScheduledEvents));
+            }
+
+            if (finish.DayOfWeek != DayOfWeek.Saturday)
+            {
+                int daysToAppendCount = DaysInOneWeek - (int)finish.DayOfWeek;
+
+                int daysRangeLength = daysCount + daysToAppendCount;
+
+                for (int day = daysCount + 1; day < daysRangeLength; day++)
+                {
+                    dayContainers.Add(GetDateContainerHtml(start.AddDays(day)));
+                }
+
+                daysCount += daysToAppendCount;
+            }
+
+            if (start.AddDays(1).DayOfWeek != DayOfWeek.Sunday)
+            {
+                int daysToPrependCount = (int)start.DayOfWeek;
+
+                for (int i = 0; i <= daysToPrependCount; i++)
+                {
+                    dayContainers.Insert(0, GetDateContainerHtml(start.AddDays(-i)));
+                }
+            }
+
+            double rowCount = (double)daysCount / DaysInOneWeek;
+
+            if (rowCount - (int)rowCount != 0)
+                rowCount = (int)rowCount + 1;
```

Comment:
Please explain this if statement.
Also, please use full if statement syntax with `{ }` curly brackets

Reply:
rowCount should be rounded to bigger value, because rowCount is a number of weeks rendered in Calendar.
ok. 9addd8f7c3c858f432346be32c39219df9043ca0